lib/systems/default.nix

```diff
@@ -384,8 +384,14 @@ let
             }.${cpu.name} or cpu.name;
             vendor_ = final.rust.platform.vendor;
           # TODO: deprecate args.rustc in favour of args.rust after 23.05 is EOL.
-          in args.rust.rustcTarget or args.rustc.config
-            or "${cpu_}-${vendor_}-${kernel.name}${optionalString (abi.name != "unknown") "-${abi.name}"}";
+          in
+            args.rust.rustcTarget or
+            args.rustc.config or (
+              # Rust uses `wasm32-wasi` rather than `wasm32-unknown-wasi`.
+              if final.isWasi
+              then "${cpu_}-wasi"
```

Comment:
Shouldn't this be wasip1 or wasip2? wasm32-wasi is [deprecated](https://blog.rust-lang.org/2024/04/09/updates-to-rusts-wasi-targets.html).

Reply:
Oh I didn't know it until now. Thanks for the info.

It seems that we do not have WASI ABI version in the platform configuration, so we cannot know if the user mean p1 or p2. I guess for compatibility, we should by default use `wasm32-wasip1`?

Reply:
We might want to go to p2 at some point, but we should definitely stick to p1 at the moment, because p2 is still tier 3.

Reply:
Updated.